segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkStorageManager.java

```diff
@@ -0,0 +1,1375 @@
+                }
+
+                if (segmentMetadata.getOwnerEpoch() > this.epoch) {
+                    throw new StorageNotPrimaryException(streamSegmentName);
+                }
+
+                segmentMetadata.setActive(false);
+
+                // Delete chunks
+                String currentChunkName = segmentMetadata.getFirstChunk();
+                ChunkMetadata currentMetadata;
+                ArrayList<String> chunksToDelete = new ArrayList<>();
+                while (currentChunkName != null) {
+                    currentMetadata = (ChunkMetadata) txn.get(currentChunkName);
+                    // Delete underlying file.
+                    chunksToDelete.add(currentChunkName);
+                    currentChunkName = currentMetadata.getNextChunk();
+                    txn.delete(currentMetadata.getName());
+                }
+
+                // Commit.
+                txn.delete(streamSegmentName);
+                txn.commit();
+
+                // Collect garbage.
+                collectGarbage(chunksToDelete);
+
+                // Update the read index.
+                cachedReadIndex.remove(streamSegmentName);
+
+                Duration elapsed = timer.getElapsed();
+                log.debug("delete - segment={} latency={}.", handle.getSegmentName(), elapsed.toMillis());
+                LoggerHelpers.traceLeave(log, "delete", traceId, handle);
+                return null;
+            } catch (StorageMetadataVersionMismatchException ex) {
+                throw ex;
+            } catch (StorageMetadataWritesFencedOutException ex) {
+                throw new StorageNotPrimaryException(streamSegmentName, ex);
+            }
+        });
+    }
+
+    /**
+     * Truncates all data in the given StreamSegment prior to the given offset. This does not fill the truncated data
+     * in the segment with anything, nor does it "shift" the remaining data to the beginning. After this operation is
+     * complete, any attempt to access the truncated data will result in an exception.
+     * <p>
+     * Notes:
+     * * Depending on implementation, this may not truncate at the exact offset. It may truncate at some point prior to
+     * the given offset, but it will never truncate beyond the offset.
+     *
+     * @param handle  A read-write SegmentHandle that points to a Segment to write to.
+     * @param offset  The offset in the StreamSegment to truncate to.
+     * @param timeout Timeout for the operation.
+     * @return A CompletableFuture that, when completed, will indicate the operation succeeded. If the operation failed,
+     * it will contain the cause of the failure. Notable exceptions:
+     * <ul>
+     * <li> StreamSegmentNotExistsException: When the given Segment does not exist in Storage.
+     * </ul>
+     */
+    @Override
+    public CompletableFuture<Void> truncate(SegmentHandle handle, long offset, Duration timeout) {
```

Comment:
method too long

Reply:
Java doesn't have out parameters . So extracting smaller methods requires creating method objects etc. So instead of making code simple it makes it unnecessarily complicated and in my opinion defeats the purpose. If there is a simple way to break this method I'll definitely do that.